client/my-sites/post-share/index.jsx

```diff
@@ -123,7 +140,26 @@ class PostShare extends Component {
 	};
 
 	sharePost = () => {
-		this.props.sharePost( this.props.siteId, this.props.postId, this.state.skipped, this.state.message );
+		const {
+			postId,
+			siteId,
+			connections,
+		} = this.props;
+		if ( this.state.scheduledDate ) {
+			const servicesToPublish = difference(
+				connections.map( connection => connection.keyring_connection_ID ),
+				this.state.skipped
+			);
+			this.props.schedulePostShareAction(
```

Comment:
If you add a `return` on this line (or after), we can avoid the `else` below, which I think would improve the readability of the function.